pyfstat/make_sfts.py

```diff
@@ -373,7 +379,50 @@ def make_cff(self):
             config_file.close()
 
     def calculate_fmin_Band(self):
-        self.fmin = self.F0 - 0.5 * self.Band
+        """
+        Set fmin and Band for the output SFTs to cover.
+
+        Either uses the user-provided Band and puts the injection in the middle,
+        or if Band==None estimates a minimal band for just the injected signal:
+        F-stat covering band plus extra bins for demod default parameters.
+        This way a perfectly matched single-template ComputeFstat analysis
+        should run through perfectly on the SFTs.
+        For any wider-band or mismatched search, the set Band manually.
+        """
+        if self.Band is None:
+            extraBins = (
+                # matching extraBinsFull in XLALCreateFstatInput():
+                # https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/_compute_fstat_8c_source.html#l00490
+                lalpulsar.FstatOptionalArgsDefaults.Dterms
+                + int(lalpulsar.FstatOptionalArgsDefaults.runningMedianWindow / 2)
+                + 1
+            )
+            logging.info(
+                "Estimating required SFT frequency range from properties"
+                " of signal to inject plus {:d} extra bins either side"
+                " (corresponding to default F-statistic settings).".format(extraBins)
+            )
+            minCoverFreq, maxCoverFreq = helper_functions.get_covering_band(
```

Comment:
I propose to put this bit in a separate method such that Isolated and Binary writers are decoupled.

E.g.

```
class Writer:

    def _calculate_Band_aux(self):
         return helper_functions.get_covering_band(
                tref=self.tref,
                tstart=self.tstart,
                tend=self.tend,
                F0=self.F0,
                F1=self.F1,
                F2=self.F2,
                F0band=0.0,
                F1band=0.0,
                F2band=0.0,
                maxOrbitAsini=0,
                minOrbitPeriod=0,
                maxOrbitEcc=0,
            )

class BinaryModulatedWriter:

    def _calculate_Band_aux(self):
        return helper_functions.get_covering_band(
                tref=self.tref,
                tstart=self.tstart,
                tend=self.tend,
                F0=self.F0,
                F1=self.F1,
                F2=self.F2,
                F0band=0.0,
                F1band=0.0,
                F2band=0.0,
                maxOrbitAsini=self.asini
                minOrbitPeriod=self.period
                maxOrbitEcc=self.ecc
            )
```

Reply:
If `get_covering_band()` had variable arguments (kwargs) for many different types of signals, I'd definitely agree. But since both it and the underlying lalpulsar function have a fixed set of arguments (fkdot+binary), I don't see a reason any future Writer children would need to deal with it much differently, and so I'd say the current getattr-with-default solution rather saves some code duplication...?

Reply:
I'm going to say "Ok" since the underlying lal-based machinery is quite unlikely to change enough in the foreseeable future as to break this bit of code.